app/src/views/IntakeProfile/index.tsx

```diff
@@ -17,15 +17,27 @@ export type InitialValues = Record<string, Values>;
 
 export const IntakeProfile = () => {
   const theme = useTheme();
+  const navigate = useNavigate();
   const toolbarHeight = Number(theme.mixins.toolbar.minHeight);
   const {profileId, groupId} = useParams();
+  const isMobile = useMediaQuery(theme.breakpoints.down('md'));
+  const [showSections, setShowSections] = useState(isMobile);
+  const [selectedItem, setSelectedItem] = useState<string | null>(
+    groupId || null,
+  );
 
   const {data: profileData} = useGetProfileQuery(
     {profileId: profileId},
     {skip: !profileId},
   );
   const {data: responsesData} = useGetResponsesQuery({userId: '1'});
 
+  useEffect(() => {
+    if (groupId) {
+      setSelectedItem(groupId);
+    }
+  }, [groupId]);
+
```

Comment:
For the selected item check, could we pass the `groupId` accessed in the params to the sidebar and do an equality check against the field group id? That way we can treat the groupId from useParams like the state variable.

Reply:
Nice catch for some optimization! Changes made.